espnet/nets/pytorch_backend/transformer/subsampling.py

```diff
@@ -8,19 +8,18 @@
 
 import torch
 
-from espnet.nets.pytorch_backend.transformer.embedding import PositionalEncoding
-
 
 class Conv2dSubsampling(torch.nn.Module):
     """Convolutional 2D subsampling (to 1/4 length).
 
     :param int idim: input dim
     :param int odim: output dim
     :param flaot dropout_rate: dropout rate
+    :param nn.Module pos_enc_class: positional encoding layer
 
     """
 
-    def __init__(self, idim, odim, dropout_rate):
+    def __init__(self, idim, odim, dropout_rate, pos_enc_class):
```

Comment:
Does this extension keep the compatibility? (Can we use the old model?)

Reply:
Actually, I want to discuss this with you in our group meeting. I added this argument because for relative position encoding, we may need to input a position vector with reverse order. @ShigekiKarita suggest I implement another class `RelPositionalEncoding` inheritance the class `PositionalEncoding`. I didn't found a better way to change the `pos_enc_class` inside the class `Conv2dSubsampling` and passed a pos enc entity to `Conv2dSubsampling` provides options for `PositionalEncoding`, `RelPositionalEncoding` or a CNN-based pos enc which shows effcient in @hiratake55 's toolkit.

Reply:
@sw005320 I mostly agree with @pengchengguo. Regarding compatibility, this PR controls sinusoidal and relative positional encoding with return values. I personally suggest that we add sinusoidal positional encoding to inputs outside `Conv2dSubsampling` as done in `RelPositionalEncoding` in order to make them simpler.